changelogs/fragments/597-openssh_keypair-generate-new-key.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - "openssh_keypair - always generate a new key pair if the private key doesn't exist. Previously, the module would fail when ``regenerate=fail`` without an existing key, contradicting the documentation."
```

Comment:
```suggestion
  - "openssh_keypair - always generate a new key pair if the private key does not exist. Previously, the module would fail when ``regenerate=fail`` without an existing key, contradicting the documentation (https://github.com/ansible-collections/community.crypto/pull/598)."
```